lib/shared/pe_status_check.rb

```diff
@@ -146,6 +150,28 @@ def self.filesystem_free(path)
     (stat.blocks_available.to_f / stat.blocks.to_f * 100).to_i
   end
 
+  #Get the maximum and current concurrent connections to Postgres
+  def self.psql_return_result(sql, psql_options = '')
+    command = %(su pe-postgres --shell /bin/bash --command "cd /tmp && #{PUP_PATHS[:server_bin]}/psql #{psql_options} --command \\"#{sql}\\"")
```

Comment:
We need to set a timeout here. Either set `PGOPTIONS='--statement-timeout=2s'` in the command like the following, which will output `ERROR:  canceling statement due to statement timeout` when it times out.
```
command = %(su pe-postgres --shell /bin/bash --command "cd /tmp && PGOPTIONS='--statement-timeout=2s' #{PUP_PATHS[:server_bin]}/psql #{psql_options} --command \\"#{sql}\\"")
```

Or use the `:timeout` option in [Facter::Core::Execution.execute](https://www.rubydoc.info/gems/facter/Facter%2FCore%2FExecution.execute). With the code below, you would need to catch the exception thrown by the timeout.

```
Facter::Core::Execution.execute(command, timeout: 2 )
```